velox/common/caching/ScanTracker.cpp

```diff
@@ -31,8 +44,7 @@ void ScanTracker::recordReference(
   if (fileGroupStats_) {
     fileGroupStats_->recordReference(fileId, groupId, id, bytes);
   }
-  std::lock_guard<std::mutex> l(mutex_);
-  auto& data = data_[id];
+  auto& data = trackingData(id);
```

Comment:
It is not thread-safe to access tracking data out of lock

It might be better to keep the current trackingData api to return a copy. And provide method to update tracking data to update the tracking stats with lock as for now. You can change mutex with rw lock to do that.

Reply:
The trackingData members are all atomic now. This allows us to not take write lock to update individual TrackingId stats.